controllers/provider-gcp/pkg/apis/gcp/validation/infrastructure.go

```diff
@@ -80,11 +79,18 @@ func ValidateInfrastructureConfig(infra *apisgcp.InfrastructureConfig, nodesCIDR
 		allErrs = append(allErrs, field.Invalid(networksPath.Child("vpc", "cloudRouter"), infra.Networks.VPC.CloudRouter, "cloud router can not be configured when the VPC name is not specified"))
 	}
 
+	if infra.Networks.VPC != nil && len(infra.Networks.VPC.Name) > 0 && infra.Networks.VPC.CloudRouter == nil {
```

Comment:
Can you make it a superior condition check `infra.Networks.VPC != nil && len(infra.Networks.VPC.Name) > 0` and then add the other two cases below it?